src/x86_parser.rs

```diff
@@ -215,16 +219,34 @@ pub fn populate_instructions(xml_contents: &str) -> anyhow::Result<Vec<Instructi
     }
 
     // provide a URL example page -----------------------------------------------------------------
-    // parse this x86 page, grab the contents of the table + the URLs they are referring to
-    // TODO Fetching fliecloutier.com may take time and sometimes it's unresponsive. Cache this
-    //      once and re-use.
-    // TODO Add a CLI argument to refresh this cache.
+    // 1. If the cache refresh option is enabled or the cache doesn't exist, attempt to fetch the
+    //    data, write it to the cache, and then use it
+    //      - parse this x86 page, grab the contents of the table + the URLs they are referring to
+    // 2. Otherwise, attempt to read the data from the cache
+    let cache_refresh = args() // replace with an actual way to check CLI args
+        .any(|arg| arg.contains("refresh"));
     let x86_online_docs = String::from("https://www.felixcloutier.com/x86/");
-    debug!(
-        "Fetching further documentation from the web -> {}...",
-        x86_online_docs
-    );
-    let body = reqwest::blocking::get(&x86_online_docs)?.text()?;
+    let mut x86_cache_path = match get_cache_dir() {
+        Ok(cache_path) => cache_path,
+        Err(e) => {
+            error!("Failed to resolve the cache file path - Error: {e}\n");
+            return Err(e);
+        }
+    };
+
+    // At this point we have a valid directory, now append the file name
+    x86_cache_path.push("x86_docs.cache");
+    let cache_exists = matches!(x86_cache_path.try_exists(), Ok(true));
+
+    let body = if cache_refresh || !cache_exists {
+        let contents = get_x86_docs_web(&x86_online_docs)?;
```

Comment:
Not 100% tied to the PR but could we catch errors at this step as well?
Not having an internet connection (and not having created the cache altogether) would result in an error.

I've made the same comment for when accessing the cache. Perhaps it would be better to wrap this whole block in a separate call, catch the errors and just return an empty body altogether in case of errors?

Reply:
I think wrapping the block in a function call makes a lot more sense!